### ISSUE TITLE
flag to check if command line was entered from visual selection

### ISSUE THREAD

**Issue by Konfekt** [vim/vim#15327]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**

It seems hard to check whether the command line was entered from visual selection (that is, the command-line is entered with `'<,'>`  prepended). 

The `ModeChanged` event does not work reliably; instead the `CursorMoved` event is more robust ([madox2/vim-ai@`46e3d5a` (#77)](https://github.com/madox2/vim-ai/pull/77/commits/46e3d5a609fdaa2ddb2af6cc5cb4e754b8b39bfa)#); however, by positioning the cursor utmost right of a text object, a visual selection can be made without cursor movement, [breaking this solution](https://github.com/madox2/vim-ai/issues/112#issuecomment-2243190390).

**Describe the solution you'd like**

Either

- make ModeChanged reliably fire the correct last and current mode, or 
- add variables that reliably track these modes, or
- add a flag for this common use case

**Comment by cposture** [comment]
I tried ModeChanged and found that after selecting the content and entering the command line, the mode change process: v, n, c; there is an unexpected n between v and c, which will prevent correct judgment @Konfekt

**Comment by Konfekt** [comment]
@cposture, yes, exactly, I ended always up finding transitions between normal modes, rendering the ModeChange event moot for this use case.

**Comment by chrisbra** [comment]
why do you need this?

**Comment by Konfekt** [comment]
For example, one would like to apply a command `:Cmd` to the character-wise visually selected range; however, `:h cmdline-ranges` are only line-wise. If one knew that the command line `:'<,'>Cmd` was entered coming from visual mode, then the exact range would be known to be given by `'<,'>`.

**Comment by Konfekt** [comment]
Here are [two](https://stackoverflow.com/questions/4316193/check-if-there-is-a-current-active-visual-selection-in-vim-from-a-function-invok?rq=3) similar [questions](https://stackoverflow.com/questions/19403155/in-custom-command-function-checking-if-current-buffer-has-sometext-selected-v/20133772#20133772) on Stackoverflow asking how to know if a visual selection is active in command-line mode.
